src/Symfony/Component/HttpKernel/Controller/LegacyArgumentResolver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Responsible for the creation of the action arguments.
+ *
+ * @deprecated This class is deprecated since 3.1 and will be removed in 4.0. Please use the ArgumentResolver::class instead.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class LegacyArgumentResolver implements ArgumentResolverInterface
+{
+    public function getArguments(Request $request, $controller)
+    {
+        // only trigger the deprecation notice if actually used, the ControllerResolver still extends this for BC reasons
+        @trigger_error(sprintf('The %s class is deprecated since 3.1 and will be removed in 4.0. Please use the %s instead.', __CLASS__, ArgumentResolver::class), E_USER_DEPRECATED);
+
+        if (is_array($controller)) {
+            $r = new \ReflectionMethod($controller[0], $controller[1]);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $r. Configured minimum length is 3.](https://www.codacy.com/app/git_9/symfony/file/2331529108/issues/source?bid=3207821&fileBranchId=3207847#l31)